src/plugin/debugger/mhpdbgc.c

```diff
@@ -1049,6 +1049,36 @@ static void mhp_dump_to_file(int argc, char * argv[])
    close(fd);
 }
 
+static const char *get_type_from_mcb(struct MCB *mcb)
+{
+  const char *dta = "Data", *env = "Environment", *inv = "Invalid";
```

Comment:
I thought in the expression `const char *dta = "Data";` that the pointer `dta` was an automatic variable so created and destroyed on enter and exit of the function, but the string constant "Data" was read only but static and available always.

Reply:
> but the string constant "Data" was read only but static and available always.

I believe this is an assumption not
backed up by the C standard.
Since the "Data" string is inside the
function, I believe it can as well be
copied on stack, even if that would
be totally silly to do.

Reply:
Thanks, I'll move them. I expect there are others I've introduced elsewhere too!

Reply:
https://stackoverflow.com/questions/1588976/do-pointers-to-string-literals-remain-valid-after-a-function-returns

2.13.4 Ordinary string literals and UTF-8 string literals are also referred to as narrow string literals. A narrow string literal has type “array of n const char”, where n is the size of the string as defined below, and has static storage duration

Seems like you are right, though.